aws_lambda_powertools/shared/user_agent.py

```diff
@@ -0,0 +1,149 @@
+import logging
+import os
+
+from aws_lambda_powertools.shared.version import VERSION
+
+powertools_version = VERSION
+inject_header = True
+
+try:
+    import botocore
+except ImportError:
+    # if botocore failed to import, user might be using custom runtime and we can't inject header
+    inject_header = False
+
+logger = logging.getLogger(__name__)
+
+EXEC_ENV: str = os.environ.get("AWS_EXECUTION_ENV", "NA")
+TARGET_SDK_EVENT: str = "request-created"
+FEATURE_PREFIX: str = "PT"
+DEFAULT_FEATURE: str = "no-op"
+HEADER_NO_OP: str = f"{FEATURE_PREFIX}/{DEFAULT_FEATURE}/{powertools_version} PTEnv/{EXEC_ENV}"
```

Comment:
```suggestion
EXEC_ENV = os.environ.get("AWS_EXECUTION_ENV", "NA")
TARGET_SDK_EVENT = "request-created"
FEATURE_PREFIX = "PT"
DEFAULT_FEATURE = "no-op"
HEADER_NO_OP = f"{FEATURE_PREFIX}/{DEFAULT_FEATURE}/{powertools_version} PTEnv/{EXEC_ENV}"
```

You don't need to annotate as string because it's inferred :) Only L17 would generally need but you provide a sentinel value which is also `str` ;)